### PR TITLE
test: added testing basics

### PR DESCRIPTION
**Pull request by benswinburne**

### CODE REVIEW

**Review by guyfedwards — COMMENTED** [review]
move tests next to their respective source files rather than a `__tests__` directory, `module-name.test.js` or `module-name.spec.js`

File: __tests__/url-verification.spec.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+describe('URL verification', () => {
+  const OLD_ENV = process.env;
+
+  beforeEach(() => {
+    process.env = { ...OLD_ENV };
+  });
+
+  afterEach(() => {
+    process.env = OLD_ENV;
+  });
```
Comment (guyfedwards) [inline __tests__/url-verification.spec.js]: Why is this needed when the `beforeEach` above exists?

**Review by guyfedwards — CHANGES_REQUESTED** [review]
not about that `__tests__` directory life. Otherwise looks dope :+1:

File: __tests__/url-verification.spec.js (hunk 1 above)
Comment (benswinburne) [reply __tests__/url-verification.spec.js]: For consistency throughout the suite of tests. `afterEach` outside of this scope would have the updated values before they're reset, global cleanup would have process.env in an unexpected state if it needs it. It's also the only place you can conceivably reset it to the right values after the last test in this describe has run if you wanted/needed to.

Mostly for safety and completeness and given it's a tremendously cheap operation it's ok i figure.

Comment (guyfedwards) [reply __tests__/url-verification.spec.js]: I believe each test is isolated so this wouldn't be an issue but not exactly impacting performance so :ok_hand:

**Comment by benswinburne** [comment]
> move tests next to their respective source files rather than a __tests__ directory, module-name.test.js or module-name.spec.js

Started them off there because initially i was going to write some integration tests but the couple i wrote in the end are unit so can move them 👍

File: __tests__/url-verification.spec.js (hunk 1 above)
Comment (benswinburne) [reply __tests__/url-verification.spec.js]: The tests are isolated per file by creating a worker pool of child processes that run tests but if you were to use --runInBand it uses a single process. Safety first, Guy. 👷

But yeah, mostly doesn't really matter

Comment (guyfedwards) [reply __tests__/url-verification.spec.js]: 🤦‍♂️

**Review by guyfedwards — APPROVED** [review]
